lib/avrora/encoder.ex

```diff
@@ -16,6 +16,32 @@ defmodule Avrora.Encoder do
     map_type: :proplist,
     record_type: :map
   }
+  @ocf_schema_full_name_pos 6
+
+  @doc """
+  Extract schema from binary Avro
+
+  ## Examples
+
+      ...> payload = <<0, 0, 0, 0, 8, 72, 48, 48, 48, 48, 48, 48, 48, 48, 45, 48,
+      48, 48, 48, 45, 48, 48, 48, 48, 45, 48, 48, 48, 48, 45, 48, 48, 48, 48, 48,
+      48, 48, 48, 48, 48, 48, 48, 123, 20, 174, 71, 225, 250, 47, 64>>
+      ...> Avrora.Encoder.extract_schema(payload)
+      {:ok, %Avrora.Schema{"full_name" => "io.confluent.Payment", "id" => 42}}
+  """
+  def extract_schema(payload) when is_binary(payload) do
+    case payload do
+      <<@registry_magic_bytes, <<id::size(32)>>, _body::binary>> ->
+        Resolver.resolve(id)
+
+      <<@object_container_magic_bytes, _::binary>> ->
+        {_, schema_info, _} = :avro_ocf.decode_binary(payload)
+        {:ok, %Avrora.Schema{full_name: elem(schema_info, @ocf_schema_full_name_pos)}}
```

Comment:
The issue here is that this schema can not be used later for the decode of the message because it contains nothing but a full name.

Reply:
What would be the required data to decode the message later?
The decoded schema used with session registry is the following
```elixir
%Avrora.Schema{
  full_name: "io.confluent.Payment",
  id: 42,
  json: "{\"namespace\":\"io.confluent\",\"name\":\"Payment\",\"type\":\"record\",\"fields\":[{\"name\":\"id\",\"type\":\"string\"},{\"name\":\"amount\",\"type\":\"double\"}]}",
  lookup_table: #Reference<0.3819334430.3431071750.144031>,
  version: nil
}
```
What we get from `:avro_ocf.decode_binary/1` is
```elixir
{:avro_record_type, "Payment", "io.confluent", "", [],
 [
   {:avro_record_field, "id", "", {:avro_primitive_type, "string", []},
    :undefined, :ascending, []},
   {:avro_record_field, "amount", "", {:avro_primitive_type, "double", []},
    :undefined, :ascending, []}
 ], "io.confluent.Payment", []}
```

Reply:
@Strech The only thing i think we can reconstruct from that is the `json` field, what about something like this? The following is just a raw idea 😄 
```elixir
defmodule Avrora.Schema.OFCParser do
  def parse(schema_info) do
    case elem(schema_info, 0) do
      :avro_record_type ->
        %{}
        |> Map.put(:namespace, elem(schema_info, 6))
        |> Map.put(:type, "record")
        |> Map.put(:fields, Enum.map(elem(schema_info, 5), &parse/1))
      :avro_record_field ->
        %{}
        |> Map.put(:name, elem(schema_info, 1))
        |> Map.put(:type, elem(elem(schema_info, 3),1))
      _ -> %{}
    end
  end
end
```

Reply:
Yeah, questionable, ideally we should be able to extract schema and put it into lookup table, later use it, but he whole idea of OCF is no need to store the schema, maybe you are right and we should just have full_name, but then we should adjust Avrora.Schema struct spec

Reply:
I would say if you can extract schema, add it to the lookup table it would be cool :+1:

Reply:
I don't understand how we can reuse the schema if added to the lookup table.
For the OCF we are not going to hit the schema registry. 
Also, when decoding we are not using the schema in ETS for OFC and even if we are using `decode/2` providing the schema name it will be ignored.

Reply:
Yep, but in terms of usage of extract schema (just separate), a full name is not a schema at all, right? So we need to put some sense in OCF format, I would say the most crucial things are:

1. Lookup table
2. Full name (needed to retrieve erlavro format from the lookup table)
3. JSON (but it's needed only to sculpt lookup table)